projects/media-viewer/src/lib/annotations/annotation-api.service.spec.ts

```diff
@@ -121,4 +121,21 @@ describe('AnnotationApiService', () => {
     req.flush(mockAnnotationSet);
   }));
 
+
+  it('gets an annotation set or creates it if it does not exist', async(() => {
+    api.getOrCreateAnnotationSet(dmDocumentId).subscribe((response) => {
+      expect(response.documentId).toBe(dmDocumentId);
+    });
+
+    const req1 = httpMock.expectOne(`/em-anno/annotation-sets/filter?documentId=${dmDocumentId}`);
+    expect(req1.request.method).toBe('GET');
+    req1.error(new ErrorEvent('Not found'), { status: 404 });
+
+    const req2 = httpMock.expectOne('/em-anno/annotation-sets');
```

Comment:
should this be a separate test?

Reply:
no it's a single test, but it simulates two requests because it first checks to see if they exist, if not it sends a second request to create a new annotation